src/docs/development/platform-integration/platform-views.md

```diff
@@ -466,18 +466,28 @@ class FLNativeViewFactory: NSObject, FlutterPlatformViewFactory {
 
 class FLNativeView: NSObject, FlutterPlatformView {
     private var _view: UIView
-
+    
     init(
         frame: CGRect,
         viewIdentifier viewId: Int64,
         arguments args: Any?,
         binaryMessenger messenger: FlutterBinaryMessenger?
     ) {
-        super.init()
         _view = UIView()
+        super.init()
     }
-
+    
     func view() -> UIView {
+        // This function returns a UIView and you can add native iOS UI here.
+        _view.backgroundColor = UIColor.blue
+        
+        let nativeLabel = UILabel()
+        nativeLabel.text = "Native text from iOS"
+        nativeLabel.textColor = UIColor.white
+        nativeLabel.textAlignment = .center
+        nativeLabel.frame = _view.frame
+        _view.addSubview(nativeLabel)
```

Comment:
we actually recommend to do this work in the constructor. is there a reason why the getter needs to do this work? cc @bparrishMines

Reply:
I've tried creating views in the constructor, but for some reason I'm unable to fetch the `UIView.frame` where `UILabel` in the sample will be placed. The output here is that the `UILabel` won't be displayed, but the background color was able to change.

```swift
init(
    frame: CGRect,
    viewIdentifier viewId: Int64,
    arguments args: Any?,
    binaryMessenger messenger: FlutterBinaryMessenger?
) {
    _view = UIView()
    super.init()
    createNativeView(view: _view)
}
```

Moved creating views into a function

```swift
func createNativeView(view _view: UIView){
    _view.backgroundColor = UIColor.blue

    let nativeLabel = UILabel()
    nativeLabel.text = "Native text from iOS"
    nativeLabel.textColor = UIColor.white
    nativeLabel.textAlignment = .center
    nativeLabel.frame = _view.frame
    _view.addSubview(nativeLabel)
}
```

Any idea what might have been the issue? And any suggestions for best practices that we can add in the docs?

Reply:
Thanks! I've moved the creation of views in `init()`